apis/cloudantv1/v1alpha1/cloudantdatabase_types.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// CloudantDatabaseParameters are the configurable fields of a CloudantDatabase.
+type CloudantDatabaseParameters struct {
+
+	// The name of the database
+	// +immutable
+	Db string `json:"db"`
+
+	// Query parameter to specify whether to enable database partitions when creating a database.
+	// +immutable
+	// +optional
+	Partitioned *bool `json:"partitioned,omitempty"`
+
+	// The number of shards in the database. Each shard is a partition of the hash value range. Default is 8, unless
+	// overridden in the `cluster config`.
+	// +immutable
+	// +optional
+	Q *int64 `json:"q,omitempty"`
+}
+
+// CloudantDatabaseObservation are the observable fields of a CloudantDatabase.
+type CloudantDatabaseObservation struct {
+
+	// Schema for database cluster information.
+	Cluster *DatabaseInformationCluster `json:"cluster,omitempty"`
+
+	// An opaque string that describes the committed state of the database.
+	CommittedUpdateSeq string `json:"committedUpdateSeq,omitempty"`
+
+	// True if the database compaction routine is operating on this database.
+	CompactRunning bool `json:"compactRunning,omitempty"`
+
+	// An opaque string that describes the compaction state of the database.
+	CompactedSeq string `json:"compactedSeq,omitempty"`
+
+	// The version of the physical format used for the data when it is stored on disk.
+	DiskFormatVersion int64 `json:"diskFormatVersion,omitempty"`
+
+	// A count of the documents in the specified database.
+	DocCount int64 `json:"docCount,omitempty"`
+
+	// Number of deleted documents.
+	DocDelCount int64 `json:"docDelCount,omitempty"`
+
+	// The engine used for the database.
+	Engine string `json:"engine,omitempty"`
+
+	// Schema for database properties.
+	Props *DatabaseInformationProps `json:"props,omitempty"`
+
+	// Schema for size information of content.
+	Sizes *ContentInformationSizes `json:"sizes,omitempty"`
+
+	// An opaque string that describes the state of the database. Do not rely on this string for counting the number of
+	// updates.
+	UpdateSeq string `json:"updateSeq,omitempty"`
+
+	// uuid in tag should be all lowercase right ??
```

Comment:
Right, json tags start always lowercase and use CamelCase when there are multiple words, so in this case it's all lowercase